kotlin/services/eventbridge/src/main/kotlin/com/kotlin/eventbridge/EventbridgeMVP.kt

```diff
@@ -0,0 +1,729 @@
+
+        val delOb = Delete {
+            objects = toDelete
+        }
+
+        val dor = DeleteObjectsRequest {
+            bucket = bucketName
+            delete = delOb
+        }
+        s3Client.deleteObjects(dor)
+
+        // Delete the S3 bucket.
+        val deleteBucketRequest = DeleteBucketRequest {
+            bucket = bucketName
+        }
+        s3Client.deleteBucket(deleteBucketRequest)
+        println("You have deleted the bucket and the objects")
+    }
+}
+
+// Delete the SNS topic.
+suspend fun deleteSNSTopic(topicArnVal: String?) {
+    val request = DeleteTopicRequest {
+        topicArn = topicArnVal
+    }
+
+    SnsClient { region = "us-east-1" }.use { snsClient ->
+        snsClient.deleteTopic(request)
+        println(" $topicArnVal was deleted.")
+    }
+}
+
+// snippet-start:[eventbridge.kotlin._delete_rule.main]
+suspend fun deleteRuleByName(ruleName: String?) {
+    val ruleRequest = DeleteRuleRequest {
+        name = ruleName
+    }
+    EventBridgeClient { region = "us-east-1" }.use { eventBrClient ->
+        eventBrClient.deleteRule(ruleRequest)
+        println("Successfully deleted the rule")
+    }
+}
+// snippet-end:[eventbridge.kotlin._delete_rule.main]
+
+// snippet-start:[eventbridge.kotlin.delete.targets.main]
+suspend fun deleteTargetsFromRule(eventRuleName: String?) {
+    // First, get all targets that will be deleted.
+    val request = ListTargetsByRuleRequest {
+        rule = eventRuleName
+    }
+
+    EventBridgeClient { region = "us-east-1" }.use { eventBrClient ->
+        val response = eventBrClient.listTargetsByRule(request)
+        val allTargets = response.targets
+
+        // Get all targets and delete them.
+        if (allTargets != null) {
+            for (myTarget in allTargets) {
+                val removeTargetsRequest = RemoveTargetsRequest {
+                    rule = eventRuleName
+                    ids = listOf(myTarget.id.toString())
+                }
+                eventBrClient.removeTargets(removeTargetsRequest)
+                println("Successfully removed the target")
+            }
+        }
+    }
+}
+// snippet-end:[eventbridge.kotlin.delete.targets.main]
+
+// snippet-start:[eventbridge.kotlin._put_event.main]
+suspend fun triggerCustomRule(email: String) {
+    val json = "{" +
+        "\"UserEmail\": \"" + email + "\"," +
+        "\"Message\": \"This event was generated by example code.\"" +
+        "\"UtcTime\": \"Now.\"" +
+        "}"
+
+    val entry = PutEventsRequestEntry {
+        source = "ExampleSource"
+        detail = json
+        detailType = "ExampleType"
+    }
+
+    val eventsRequest = PutEventsRequest {
+        this.entries = listOf(entry)
+    }
+
+    EventBridgeClient { region = "us-east-1" }.use { eventBrClient ->
+        eventBrClient.putEvents(eventsRequest)
+    }
+}
+// snippet-end:[eventbridge.kotlin._put_event.main]
+
+suspend fun updateCustomRuleTargetWithTransform(topicArn: String?, ruleName: String?) {
```

Comment:
Should there be a tag on this method so it can be included in the metadata? In the other languages we have included it.